File: day01/pascal/part1.pas
{$mode objfpc}{$H+}{$J-}

program Part1;

uses
  ArrayUtils, JabbaTypes, JabbaUtils, sysutils;


procedure Main();
var
  fname, line: String;
  lines, parts: TStringArray;
  col1, col2: TIntArray;
  idx: Integer = 0;
  total: Integer = 0;
begin
  // fname := 'example.txt';
  fname := 'input.txt';
  lines := Readlines(fname);
  col1 := [];
  col2 := [];
  for line in lines do
    begin
      parts := PySplit(line);
      SetLength(col1, Length(col1) + 1);
      SetLength(col2, Length(col2) + 1);
      col1[idx] := StrToInt(parts[0]);
      col2[idx] := StrToInt(parts[1]);
      idx += 1;
    end;
  SimpleSort(col1);
  SimpleSort(col2);
  for idx := Low(col1) to High(col1) do
    total += Abs(col1[idx] - col2[idx]);
  WriteLn(total);
end;

//---------------------------------------------------------------------------

begin
  WriteLn('---');
  Main();
end.

File: day01/python/part1.py
#!/usr/bin/env python3

import helper


def main() -> None:
    # lines: list[str] = helper.read_lines("example.txt")
    lines: list[str] = helper.read_lines("input.txt")

    first: list[int] = []
    second: list[int] = []

    for line in lines:
        parts = line.split()
        first.append(int(parts[0]))
        second.append(int(parts[1]))
    #

    first.sort()
    second.sort()

    result = 0
    for a, b in zip(first, second):
        result += abs(a - b)
    #
    print(result)


##############################################################################

if __name__ == "__main__":
    main()
